zerver/openapi/zulip.yaml

```
@@ -5265,6 +5254,131 @@ paths:
+            after it was sent.
+          required: true
+          example: 2
+        - name: type
+          in: query
+          description: |
+            The type of scheduled message to be sent. `"direct"` for a direct
+            message and `"stream"` for a stream message.
+
+            When updating the type of the scheduled message, the `to` parameter
+            is required. And, if updating the type of the scheduled message to
+            `"stream"`, then the `topic` parameter is also required.
+
+            In Zulip 7.0 (feature level 174), `"direct"` was added as the
+            preferred way to indicate the type of a direct message, deprecating
+            the original `"private"`. While `"private"` is supported for
+            scheduling direct messages, clients are encouraged to use to the
+            modern convention because support for `"private"` may eventually
+            be removed.
+          schema:
+            type: string
+            enum:
+              - direct
+              - stream
+              - private
+          example: stream
+        - name: to
+          in: query
+          description: |
+            The scheduled message's tentative target audience.
+
+            For stream messages, the integer ID of the stream.
+            For direct messages, a list containing integer user IDs.
+
+            Required when updating the `type` of the scheduled message.
+          content:
+            application/json:
+              schema:
+                oneOf:
+                  - type: integer
+                  - type: array
+                    items:
+                      type: integer
+                    minLength: 1
+              example: 8
+        - name: content
+          in: query
+          description: |
+            The updated content of the scheduled message.
+
+            Clients should use the `max_message_length` returned by the
+            [`POST /register`](/api/register-queue) endpoint to determine
+            the maximum message size.
+          schema:
+            type: string
+          example: Hello
+        - name: topic
+          in: query
+          description: |
+            The updated topic of the scheduled message.
+
+            Required when updating the `type` of the scheduled message to
+            `"stream"`.
+
+            Clients should use the `max_topic_length` returned by the
+            [`POST /register`](/api/register-queue) endpoint to determine
+            the maximum topic length.
+          schema:
+            type: string
+          example: Castle
+        - name: scheduled_delivery_timestamp
+          in: query
+          schema:
+            type: integer
+          description: |
+            The UNIX timestamp for when the message will be sent,
+            in UTC seconds.
+
+            Required when updating a scheduled message that the server
+            failed to send. See API changelog feature level 181 note for
```

Perhaps:
```suggestion
            Required when updating a scheduled message that the server
            has already tried and failed to send. See API changelog feature level 181 note for
```

so the emphasis is more on the "has already tried" rather than the "failed". I think that makes it easier to see what this requirement is about, because the point is that the old scheduled time is in the past and so we need a new one.